common/system/tasks/buzzer_heartbeat.cpp

```diff
@@ -1,25 +1,59 @@
-
 #include "buzzer_heartbeat.h"
 
-#include "hardware_manager.h"
-
 static uint32_t last_time = 0;
-static bool isOn;
+// static bool isOn;
 static int buzzerId_ = 0;
+BuzzerState state = BuzzerState::PYRO;
 
 void buzzerHeartbeat_setBuzzer(int buzzerId) { buzzerId_ = buzzerId; }
 
+void pyroSoundOff(bool pyroContCheck, int& pyroCounter);
+
 // Alternates the buzzer between on and off states if a certain period of time
 // has passed (500ms)
-void buzzerHeartbeat_tick() {
+void buzzerHeartbeat_tick(BuzzerState state) {
+  static int pyroCounter = 0;
+  bool pyroContCheck;
+
   if ((hm_millis() - last_time) > 500) {
+    SensorData_s* sensorData = hm_getSensorData();
+
+    switch (state) {
+      case BuzzerState::PYRO:
+        if (pyroCounter >= NUM_PYRO_CONT) {
+          pyroCounter = 0;
+        }
+        pyroContCheck = sensorData->pyroContData[pyroCounter];
+        pyroSoundOff(pyroContCheck, pyroCounter);
+        break;
+      default:
+        break;
+    }
     last_time = hm_millis();
-    if (isOn) {
-      hm_buzzerSetFrequency(buzzerId_, 600);
-      hm_buzzerStart(buzzerId_);
+  }
+}
+
+void pyroSoundOff(bool pyroContCheck, int& pyroCounter) {
+  static bool on = false;
```

Comment:
Would putting these static variables into a class make more sense? It's a good chance to refactor the whole buzzer task to be a class imo

Reply:
I feel like refactoring it into a class would take a lot more decisions about the functionality that we want the buzzer to have. For now maybe update this one, merge it, then work on the refactoring?